src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/MutableObjectModelBinder.cs

```diff
@@ -358,104 +374,6 @@ protected virtual object GetModel(ModelBindingContext bindingContext)
-                                    CanUpdateProperty(propertyMetadata));
-        }
-
-        private static Func<ModelBindingContext, string, bool> GetPropertyFilter()
-        {
-            return (ModelBindingContext context, string propertyName) =>
-            {
-                var modelMetadataPredicate = context.ModelMetadata.PropertyBindingPredicateProvider?.PropertyFilter;
-
-                return
-                    (context.PropertyFilter == null || context.PropertyFilter(context, propertyName)) &&
-                    (modelMetadataPredicate == null || modelMetadataPredicate(context, propertyName));
-            };
-        }
-
-        internal static PropertyValidationInfo GetPropertyValidationInfo(ModelBindingContext bindingContext)
-        {
-            var validationInfo = new PropertyValidationInfo();
-
-            foreach (var propertyMetadata in bindingContext.ModelMetadata.Properties)
-            {
-                var propertyName = propertyMetadata.PropertyName;
-
-                if (!propertyMetadata.IsBindingAllowed)
-                {
-                    // Nothing to do here if binding is not allowed.
-                    validationInfo.SkipProperties.Add(propertyName);
-                    continue;
-                }
-
-                if (propertyMetadata.IsBindingRequired)
-                {
-                    validationInfo.RequiredProperties.Add(propertyName);
-                }
-            }
-
-            return validationInfo;
-        }
-
-        // Internal for testing.
-        internal void ProcessResults(
-            ModelBindingContext bindingContext,
-            IDictionary<ModelMetadata, ModelBindingResult> results)
-        {
-            var metadataProvider = bindingContext.OperationBindingContext.MetadataProvider;
-            var metadata = metadataProvider.GetMetadataForType(bindingContext.ModelType);
-
-            var validationInfo = GetPropertyValidationInfo(bindingContext);
-
-            // Eliminate provided properties from RequiredProperties; leaving just *missing* required properties.
-            var boundProperties = results.Where(p => p.Value.IsModelSet).Select(p => p.Key.PropertyName);
-            validationInfo.RequiredProperties.ExceptWith(boundProperties);
-
-            foreach (var missingRequiredProperty in validationInfo.RequiredProperties)
-            {
-                var propertyMetadata = metadata.Properties[missingRequiredProperty];
-                var propertyName = propertyMetadata.BinderModelName ?? missingRequiredProperty;
-                var modelStateKey = ModelNames.CreatePropertyModelName(bindingContext.ModelName, propertyName);
-
-                bindingContext.ModelState.TryAddModelError(
-                    modelStateKey,
-                    bindingContext.ModelMetadata.ModelBindingMessageProvider.MissingBindRequiredValueAccessor(
```

Comment:
This was a bug that I fixed. It should use `propertyMetadata.ModelBindingMessageProvider`